epxresults/job.py

```diff
@@ -110,6 +119,67 @@ def __init__(self, **kwargs) -> None:
         self.job_id = self._return_job_id(**kwargs)
         self.conditions, self.global_variables = self._parse_vars()
         self._snapshot_map = self._set_snapshot_map()
+        self._results = self._fred_results(**kwargs)
+
+    def _fred_results(self, **kwargs) -> Path:
+        """
+        The FRED results directory associated with the job
+
+        Notes
+        -----
+        If no results directory is associated with the job, `None` will be
+        returned.
+        """
+
+        if 'PATH_TO_JOB' in kwargs.keys():
+            root = Path(self.path_to_job).parent
+            while root != Path('/'):
+                if _is_valid_results_directory(root):
+                    return root
+                else:
+                    root = root.parent
+            return None
+        else:
+            return Path(_path_to_results(**kwargs))
+
+    def delete(self, force=False, verbose=False):
+        """
+        Delete the FRED job
+        """
+
+        if force is False:
+            permission_to_delete = None
+            q = (f"Are you sure you want to delete job {self.job_key}?")
+            while permission_to_delete is None:
+                reply = str(input(q + ' (y/n): ')).lower().strip()
+                if reply[0] == 'y':
+                    permission_to_delete = True
+                if reply[0] == 'n':
+                    permission_to_delete = False
+        else:
+            permission_to_delete = True
+
+        if permission_to_delete is False:
+            msg = (f"job {self.job_key} could not be deleted.")
+            raise RuntimeError(msg)
+        elif self._results is None:
+            shutil.rmtree(self.path_to_job)
+            if verbose:
+                print(f"deleting {self.path_to_job}.")
```

Comment:
Under what circumstances might `self._results` be `None` while `self.path_to_job` isn't? What does `self._results` represent?

Reply:
You can instantiate a job from a valid job result directory, even if the job result directory isn't within a valid FRED results directory.

Reply:
I should say, that the `_results` attribute store the path to the FRED results directory which contains the job results.  This will often be `$FRED_RESULTS`.

Reply:
Ah, I see. Fair enough!